Providers/SDWebThumbnails.swift

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Storage
+
+/**
+* SDWebImage implementation of the Thumbnails protocol.
+*/
+public class SDWebThumbnails: Thumbnails {
+    private let cache: SDImageCache
+
+    required public init(files: FileAccessor) {
+        self.cache = SDImageCache(namespace: "thumbnails")
+    }
+
+    public func clear(complete: ((success: Bool) -> Void)?) {
+        cache.clearDisk()
+        cache.clearMemory()
+
+        dispatch_async(dispatch_get_main_queue()) {
+            complete?(success: true)
+            return
+        }
+    }
+
+    public class func getKey(url: NSURL) -> String {
+        return getKey(url.absoluteString!)
+    }
+
+    public class func getKey(url: String) -> String {
+        return "thumbnail://\(url.sha1.hexEncodedString)"
+    }
+
+    public func get(url: NSURL, complete: (thumbnail: Thumbnail?) -> Void) {
+        var thumbnail: Thumbnail? = nil
+        let key = SDWebThumbnails.getKey(url)
+        if let img = cache.imageFromMemoryCacheForKey(key) {
```

Comment:
The docs for `imageFromDiskCacheForKey` say:

> Query the disk cache synchronously after checking the memory cache.

So it sounds like we should use just `imageFromDiskCacheForKey` without checking both ourselves.